subsys/bluetooth/mesh/light_ctrl_srv.c

```diff
@@ -1300,26 +1314,21 @@ static void onoff_set(struct bt_mesh_onoff_srv *onoff,
 {
 	struct bt_mesh_light_ctrl_srv *srv =
 		CONTAINER_OF(onoff, struct bt_mesh_light_ctrl_srv, onoff);
-
-	const struct bt_mesh_model_transition *transition =
-		set->transition->time ? set->transition : NULL;
 	enum bt_mesh_light_ctrl_srv_state prev_state = srv->state;
 
 	if (set->transition->delay) {
 		delayed_set(srv, set->transition, set->on_off);
 	} else if (set->on_off) {
-		turn_on(srv, transition, false);
+		turn_on(srv, set->transition, false);
 	} else {
-		turn_off(srv, transition, false);
+		turn_off(srv, set->transition, false);
```

Comment:
This breaks the fallback to the configured transition time in `turn_on` and `turn_off`. IIRC the transition will never be null here.

Reply:
Hmm, ok, I initially thought that rules for the transition time for Generic OnOff state should be applied here, but after reading this part of the spec (which got more clarification), seems like here we should either use the transition time provided in the message or the default one for Light LC:
```
If the transition is started as a result of receiving the Light LC Light OnOff Set or Light LC Light OnOff Set Unacknowledged message (see Section 6.5.1.6.2), or as a result of receiving the Generic OnOff Set or Generic OnOff Set Unacknowledged message, changing the bound Generic OnOff state (see Section 6.2.3.3.1), and the Transition Time field is present in this message, it shall be used as the Transition Time.
```

However, Generic OnOff Server will always provide transition time set to either one that was provided in the message or to Generic Default Transition Time state: https://github.com/nrfconnect/sdk-nrf/blob/master/subsys/bluetooth/mesh/gen_onoff_srv.c#L81-L85 So if the default transition time is set in Generic Default Transition Time Server (greater than 0), we will still use this time, but not the default one for Light LC.